test/source_generation_test.py

```diff
@@ -142,3 +153,57 @@ class TestSelfReference(TestCase):
     """Circular references, including self references, aren't supported at this time"""
     def test_circular_dependency_triggers_error(self):
         self.assertRaises(SourceGenerationError, classes_source, [SelfReferencingRecord])
+
+
+@no_auto_store()
+class ChildWithOwnNameSpace(Record):
+    _namespace = "test.pyschema_test_child"
+    a = Integer()
+
+
+@no_auto_store()
+class ChildWithSameNamespace(Record):
+    _namespace = "pyschema_test_parent"
+
+
+@no_auto_store()
+class ParentWithNameSpace(Record):
+    _namespace = "pyschema_test_parent"
+    child = SubRecord(ChildWithOwnNameSpace)
+    other = SubRecord(ChildWithSameNamespace)
+
+
+class TestPythonPackageGeneration(TestCase):
+    def setUp(self):
+        self.tmp_path = tempfile.mkdtemp()
+
+    def get_file_content(self, relative_path):
+        return open(os.path.join(self.tmp_path, relative_path)).read()
+
+    def tearDown(self):
+        shutil.rmtree(self.tmp_path)
+
+    def test_package_generation_no_namespace(self):
+        to_python_package([Parent], self.tmp_path)
+        src = self.get_file_content("__init__.py")
+        expected_src = header_source() + "\n" + TestSubRecord.correct
+        self.assertEquals(src, expected_src)
+
+    def test_package_generation_namespace(self):
+        to_python_package([FooRecord], self.tmp_path)
+        src = self.get_file_content("my/foo/bar.py")
+        expected_src = header_source() + "\n" + TestSourceConversion.correct
+        self.assertEquals(src, expected_src)
+        #TODO, verify __init__ files
```

Comment:
Creating __init__.py files in all the folders so that we build a valid package strucute